### ISSUE TITLE
logical consistency

### ISSUE THREAD

**Issue by StephanPan** [HaiyuWu/LogicalConsistency#1]
Directly truncating prediction value with 0.5 is kinda confusing me, why don't you use softmax first?

**Comment by HaiyuWu** [comment]
Thanks for this valuable question!
"Directly truncating prediction value with 0.5" is not a correct implementation to get the binary predictions when the loss function is binary cross entropy. 
"why don't you use softmax first?", I believe what you want to say is sigmoid. 
We will fix the code and update the correct results soon.
------------------------------------------------------------------
FYI, I have a set of results for using bce-moon and bce loss with the corrected code, the logical consistency is still a problem.

**Comment by StephanPan** [comment]
Thanks for your reply! Maybe the logical consistency comes from the crossover between different dimensions. A prossible way may be merging the beard length into other dimensions like setting long mustache, long chin area alternatives or something like these.

**Comment by HaiyuWu** [comment]
We have tested this approach but it does not work on logical consistency. Maybe some other strategies need to be added to boost it.
